catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,705 @@
+        switch (type) {
+        case AttributeDefinition.BOOLEAN:
+            return Boolean.valueOf(value);
+        case AttributeDefinition.BYTE:
+            return Byte.valueOf(value);
+        case AttributeDefinition.DOUBLE:
+            return Double.valueOf(value);
+        case AttributeDefinition.CHARACTER:
+            return value.toCharArray()[0];
+        case AttributeDefinition.FLOAT:
+            return Float.valueOf(value);
+        case AttributeDefinition.INTEGER:
+            return Integer.valueOf(value);
+        case AttributeDefinition.LONG:
+            return Long.valueOf(value);
+        case AttributeDefinition.SHORT:
+            return Short.valueOf(value);
+        case AttributeDefinition.PASSWORD:
+        case AttributeDefinition.STRING:
+        default:
+            return value;
+        }
+    }
+
+    protected BundleContext getBundleContext() {
+        return FrameworkUtil.getBundle(this.getClass())
+                .getBundleContext();
+    }
+
+    private void updateRegistrySourceConfigurations() {
+        try {
+            List<Metacard> metacards = federationAdminService.getRegistryMetacards();
+
+            for (Metacard metacard : metacards) {
+                try {
+                    updateRegistryConfigurations(metacard, false);
+                } catch (InvalidSyntaxException | ParserException | IOException e) {
+                    LOGGER.error(
+                            "Unable to update registry configurations. Registry source configurations won't be updated for metacard id: {}",
+                            metacard.getId());
+                }
+            }
+
+        } catch (FederationAdminException e) {
+            LOGGER.error(
+                    "Error getting registry metacards. Registry source configurations won't be updated.");
+        }
+    }
+
+    public void setActivateConfigurations(boolean activateConfigurations) {
+        if (!(activateConfigurations == this.activateConfigurations)) {
+            this.activateConfigurations = activateConfigurations;
+
+            if (!activateConfigurations || preserveActiveConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setPreserveActiveConfigurations(boolean preserveActiveConfigurations) {
+        if (!(preserveActiveConfigurations == this.preserveActiveConfigurations)) {
+            this.preserveActiveConfigurations = preserveActiveConfigurations;
+
+            if (preserveActiveConfigurations || !activateConfigurations) {
+                return;
+            }
+
+            updateRegistrySourceConfigurations();
+        }
+    }
+
+    public void setCleanUpOnDelete(boolean cleanUpOnDelete) {
+        this.cleanUpOnDelete = cleanUpOnDelete;
+    }
+
+    public void setBindingTypeFactoryPid(List<String> bindingTypeFactoryPid) {
+        bindingTypeToFactoryPidMap = new HashMap<>();
```

Comment:
Is this method only called at startup? If not, we may actually have some threading problems since we're accessing it in `updateRegistryConfigurations()` from a potentially different thread.